src/ngraph/builder/quantization/quantized_linear_matmul.cpp

```diff
@@ -0,0 +1,138 @@
+#include "ngraph/op/experimental/quantized_dot.hpp"
+#include "ngraph/op/multiply.hpp"
+#include "ngraph/op/quantize.hpp"
+#include "ngraph/op/reshape.hpp"
+#include "ngraph/type/element_type.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+namespace ngraph
+{
+    namespace builder
+    {
+        namespace quantization
+        {
+            bool check_zero_point(shared_ptr<op::Constant> input)
+            {
+                bool is_zero = false;
+                if (input->get_element_type() == element::i8)
+                {
+                    auto input_zero_value = input->get_vector<int8_t>();
+                    is_zero = all_of(input_zero_value.begin(), input_zero_value.end(), [](int i) {
+                        return i == 0;
+                    });
+                }
+                else
+                {
+                    auto input_zero_value = input->get_vector<uint8_t>();
+                    is_zero = all_of(input_zero_value.begin(), input_zero_value.end(), [](int i) {
+                        return i == 0;
+                    });
+                }
+                return is_zero;
+            }
+
+            // TODO: this code is falling back to fp32 dot
+            //       1) add support in reference kernel for zero point
+            shared_ptr<Node> QuantizedLinearMatmul(const shared_ptr<Node>& input0,
+                                                   const shared_ptr<Node>& input1,
+                                                   const shared_ptr<Node>& input0_scale,
+                                                   const shared_ptr<Node>& input0_zero_point,
+                                                   const shared_ptr<Node>& input1_scale,
+                                                   const shared_ptr<Node>& input1_zero_point,
+                                                   const shared_ptr<Node>& output_scale,
+                                                   const shared_ptr<Node>& output_zero_point)
+            {
+                auto input0_zero = dynamic_pointer_cast<ngraph::op::Constant>(input0_zero_point);
+                auto input1_zero = dynamic_pointer_cast<ngraph::op::Constant>(input1_zero_point);
+                auto output_zero = dynamic_pointer_cast<ngraph::op::Constant>(output_zero_point);
+
+                auto reshaped_input1 = input1;
+                // Check if zero point is constant and zero
+                if (input0_zero != nullptr && input1_zero != nullptr && output_zero != nullptr &&
+                    check_zero_point(input0_zero) && check_zero_point(input1_zero) &&
+                    check_zero_point(output_zero))
+                {
+                    auto requantization_scale = (input0_scale * input1_scale) / output_scale;
+
+                    // Since u8u8 goes to ref reshape the input1 so that the ref dot can do a matmul
+                    // which requires [m, n] * [n, k] order where as mkldnn requires [m, n] * [k, n]
```

Comment:
This is the wrong fix, in my opinion.  We should document the API as [m, n] * [n, k] and have ONNX importer call it with those semantics.  ONNX should be doing this reshape, not the builder.  What if happens now if someone passes in [m, n] * [n, k]?  Input 1 would be reshaped and fail.

Reply:
For QuantizedDot we expect the bridges to pass [m, n] * [k, n] and thats how the output shape is set and all the checks are done based on that assumption. maybe @jbobba can comment more. We need to decide on one standard.

Reply:
Reshaping based on element type is simply not correct.  The builder should enforce a policy on the shape of its inputs.  In my opinion, the standard should be [m,n] * [n,k] as with the non quantized dot operator.  The caller then has the responsibility to conform it's inputs to that convention.  

Alternatively, the builder could have a more flexible policy wherein either [m,n] * [n,k] OR [m,n] * [k,n] are accepted.  The builder should then reshape inputs based on their shapes, not their element types.

There is a fundamental assumption here that callers of this builder want to run on either the CPU or on the ref kernel in the case where mkldnn has no support.  This may be the case now but it is not the case forever.  Builders and core ops need a more general implementation.